R/rma-tidiers.R

```diff
@@ -235,9 +235,9 @@ augment.rma <- function(x, ...) {
   pred <- as.data.frame(pred)
 
   # fix names
-  names(pred)[1:4] <- c(".fitted", ".se.fit", ".conf.low", ".conf.high")
-  credible_intervals <- names(pred) %in% c("cr.lb", "cr.ub")
-  names(pred)[credible_intervals] <- c(".cred.low", ".cred.high")
+  names(pred)[1:4] <- c(".fitted", ".se.fit", ".conf.lower", ".conf.upper")
```

Comment:
Since these don't end up being fully internally consistent, I'd prefer we stick with the current column naming so as to avoid breaking changes.

Reply:
The developer of metafor has acknowledged that "cred"/"credibility" was not an accurate label for the column (they are prediction intervals) and has changed its column labels from "ci" to "pi". These values are definitely not credibility intervals, so the `.cred.low` and `.cred.high` labels should at least change to `pred.low` and `.pred.high` to match what the values actually are and the metafor API.

Reply:
After that change to remove the inaccurate and misleading label on the prediction intervals, my though was that using the same "lower"/"upper" terminology as the other augment methods made sense.

Reply:
Would it be possible to label these just `.lower` and `.upper`? More comfortable with the potential breaking change if we're being internally consistent.

Reply:
There are two sets of intervals returned from `predict.rma()`—confidence intervals ("ci.lb", "ci.ub") and prediction intervals ("pi.lb", "pi.ub"). Unlike `predict.lm()`, `predict.rma()` always returns both. How should we handle that?

Reply:
Ah, right! Yeah, this is tough. Not sure if this was where your hesitancy came from originally, @alexpghayes.

Reply:
I see two possible options here:

1. The path moving forward could be: intervals are always just labeled `.lower` and `.upper` if there is only one set. They are prepended with a consistent prefix (`.conf` for confidence, `.pred` for prediction, `.cred` for credible) if and only if the augment function returns multiple intervals at once. That ensures both consistency and accuracy in labeling (vs. the status quo when the `.conf.low` label is inaccurate in augment.lm when `interval="prediction"`).
1. Only return one interval from augment.rma() rather than two. That would be easy enough and make the behavior more consistent with other augment/predict methods.

Reply:
On further review, this is a problem less than I actually thought. It's only an issue with the `rma.mv` class in metafor, which is fairly rarely used. The most common class `rma.uni` always returns prediction intervals the way `augment.rma` works. `rma.mh` and `rma.peto` only compute confidence intervals. 

So, with that in mind, I suggest we use `.lower` and `.upper` consistently. Then, for `rma.mv` only return the prediction interval (this makes the behavior consistent with rma.uni and is generally most appropriate in meta-analysis). We could add an `interval` argument to choose confidence instead.

Reply:
Okay, I've implemented that. Now, all augment methods return `.lower` and `.upper` consistently.